.github/workflows/codeql.yml

```diff
@@ -0,0 +1,104 @@
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '26 19 * * 5'
+  repository_dispatch:
+    types:
+      - test-command
+
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
+        # CodeQL supports [ 'c-cpp', 'csharp', 'go', 'java-kotlin', 'javascript-typescript', 'python', 'ruby', 'swift' ]
+        # Use only 'java-kotlin' to analyze code written in Java, Kotlin or both
+        # Use only 'javascript-typescript' to analyze code written in JavaScript, TypeScript or both
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - uses: actions/setup-go@v4
+      with:
+        go-version: '1.21.0'
+        cache: true
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+        
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+
+        # For more details on CodeQL's query packs, refer to: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+        # queries: security-extended,security-and-quality
+
+    - name: Build Application
+      run: |
+        echo "Building auth-service"
+        go build ./cmd/auth-service
+
+        echo "Building crd-replicator"
+        go build ./cmd/crd-replicator
+
+        echo "Building controler-manager"
+        go build ./cmd/liqo-controller-manager
+
+        echo "Building liqoctl"
+        go build ./cmd/liqoctl
+
+        echo "Building liqonet"
+        go build ./cmd/liqonet
+
+        echo "Building metric-agent"
+        go build ./cmd/metric-agent
+
+        echo "Building telemetry"
+        go build ./cmd/telemetry
+
+        echo "Building uninstaller"
+        go build ./cmd/uninstaller
+
+        echo "Building virtual-kubelet"
+        go build ./cmd/virtual-kubelet
```

Comment:
Can we make this more maintainable? For instance, building everything we find in the cmd folder